src/server.c

```diff
@@ -2785,6 +2785,16 @@ void initServer(void) {
     server.client_mem_usage_buckets = NULL;
     resetReplicationBuffer();
 
+    if (server.maxmemory) {
+        if (server.maxmemory < server.maxmemory_reserved) {
```

Comment:
Should we fail the "=" case too? otherwise, it effectively disables `maxmemory` I think. We should add a test case too.

```suggestion
        if (server.maxmemory <= server.maxmemory_reserved) {
```

Reply:
update "<" to "<=" i think no hurt.  It guarantees at least some keys can be stored in the server.  If we allow server.maxmemory == server.maxmemory_reserved, it means the available memory is 0 (it makes no sense)